tests/host_callback_to_tf_test.py

```diff
@@ -0,0 +1,256 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An example of using host_callback.call to invoke on the host functions
+written in Tensorflow. The interesting aspect here is how we can differentiate
+through the outside computation, using tf.GradientTape on the host.
+
+This is separate from host_callback_test because it needs a TF dependency.
+"""
+from typing import Callable
+import unittest
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+from jax import api
+from jax.config import config
+from jax import numpy as jnp
+from jax import test_util as jtu
+from jax.experimental import host_callback as hcb
+
+try:
+  import tensorflow as tf
+except ImportError:
+  tf = None
+
+config.parse_flags_with_absl()
+FLAGS = config.FLAGS
+
+
+def call_tf_no_ad(tf_fun: Callable, arg, *, result_shape):
+  """The simplest implementation of calling to TF, without AD support.
+
+  We must use hcb.call because the TF invocation must happen outside the
+  JAX staged computation."""
+
+  def tf_to_numpy(t):
+    return t.numpy() if isinstance(t, tf.Tensor) else t
```

Comment:
Maybe add a comment noting that ideally we would convert directly on device (e.g., with DLpack) rather than using NumPy?

Reply:
I did not realize that `t.numpy()` would copy the array. I have changed the code to use np.array(memoryview(t)) instead, and added a comment.